code/modules/mob/living/carbon/inventory.dm

```diff
@@ -172,9 +172,15 @@
 		return
 	visible_message("<span class='notice'>[src] is offering [receiving]</span>", \
 					"<span class='notice'>You offer [receiving]</span>", null, 2)
-	for(var/mob/living/carbon/C in orange(1, src))
+	for(var/mob/living/carbon/C in orange(1, src)) //Fixed that, now it shouldn't be able to give benos stunbatons and IDs
 		if(!CanReach(C))
 			continue
+
+		if(istype(C, /mob/living/carbon/alien))
+			var/mob/living/carbon/alien/A = C
+			if(!A.has_fine_manipulation)
+				continue
```

Comment:
You should probably check for `can_hold_items()` on all mobtypes instead, it will include the xeno fix as well as any other similar cases that might spring up later